airflow/utils/pydantic.py

```diff
@@ -24,19 +24,15 @@
 
 from __future__ import annotations
 
+from importlib.metadata import PackageNotFoundError, version
 
-def is_pydantic_2_installed() -> bool:
-    import sys
+from pkg_resources import parse_version
 
-    from packaging.version import Version
 
-    if sys.version_info >= (3, 9):
-        from importlib.metadata import distribution
-    else:
-        from importlib_metadata import distribution
+def is_pydantic_2_installed() -> bool:
     try:
-        return Version(distribution("pydantic").version) >= Version("2.0.0")
-    except ImportError:
+        return parse_version(version("pydantic")).major == 2
+    except PackageNotFoundError:
```

Comment:
A unit test for the correct exception type might be an overkill, but I would add a test just so it reminds us that the string provided to `version` needs to be the "pypi name" and not the "import name" (i.e. `apache-airflow` would work whereas `airflow` wouldn't).

Reply:
That is not an our method . Why do you want unit test external method ? I guess it already has unit tests in stdlib. Can you check there ? I don't think we shou unit tests every single method call in stdlib - I believe this is what you suggest ?